pwa-wp.php

```diff
@@ -41,4 +41,40 @@ function pwawp_init() {
 
 	$wp_web_app_manifest = new WP_Web_App_Manifest();
 	$wp_web_app_manifest->init();
+
+	// These could be in ABSPATH . WPINC . '/script-loader.php' file.
+	/** WordPress Service Workers Class */
+	require PWAWP_PLUGIN_DIR . '/wp-includes/class.wp-service-workers.php';
+
+	/** WordPress Scripts Functions */
+	require PWAWP_PLUGIN_DIR . '/wp-includes/functions.wp-service-workers.php';
+
+	add_action( 'wp_print_footer_scripts', 'wp_print_service_workers' );
+
+	// Alternative for this could be in wp-includes/functions.php.
+	add_action( 'template_redirect', 'pwawp_maybe_display_sw_script' );
+
+	add_action( 'init', 'pwawp_add_sw_rewrite_rules' );
+}
+
+/**
+ * Register rewrite rules for Service Workers.
+ */
+function pwawp_add_sw_rewrite_rules() {
+	add_rewrite_tag( '%wp_service_worker%', '(0|1)' );
+	add_rewrite_tag( '%scope%', '([^&]+)' );
+	add_rewrite_rule( '^wp-service-worker.js?', 'index.php?wp_service_worker=1', 'top' );
+}
+
+/**
+ * If it's a service worker script page, display that.
+ */
+function pwawp_maybe_display_sw_script() {
+
+	if (
+		true === filter_var( get_query_var( 'wp_service_worker' ), FILTER_VALIDATE_BOOLEAN ) &&
```

Comment:
The `filter_var()` functions aren't used in core. So this can just check if `get_query_var( 'wp_service_worker' )` is not empty.